specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2021-08-01/managedClusters.json

```diff
@@ -5082,6 +5346,76 @@
           "description": "This is the ARM ID of the source object to be used to create the target object."
         }
       }
+    },
+    "SnapshotListResult": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Snapshot"
+          },
+          "description": "The list of snapshots."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The URL to get the next set of snapshot results.",
+          "readOnly": true
+        }
+      },
+      "description": "The response from the List Snapshots operation."
+    },
+    "Snapshot": {
+      "type": "object",
+      "properties": {
+        "systemData": {
+          "$ref": "#/definitions/SystemData",
+          "readOnly": true,
+          "description": "The system metadata relating to this snapshot."
+        },
+        "properties": {
+          "description": "Properties of a snapshot.",
+          "$ref": "#/definitions/SnapshotProperties",
+          "x-ms-client-flatten": true
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "description": "An agent pool snapshot resource."
+    },
+    "SnapshotProperties": {
+      "type": "object",
+      "properties": {
+        "creationData": {
+          "$ref": "#/definitions/CreationData",
+          "description": "CreationData to be used to specify the source agent pool resource ID to create this snapshot."
+        },
+        "snapshotType": {
+          "$ref": "#/definitions/SnapshotType"
+        }
+      },
+      "description": "Properties used to configure an agent pool snapshot."
+    },
+    "SnapshotType": {
+      "type": "string",
+      "default": "Nodepool",
+      "enum": [
+        "Nodepool"
```

Comment:
As mentioned above, shouldn't this be `AgentPool` since that's what we call it elsewhere in the public API, including in the routes? (`/agentpools`).

You even call it `an agent pool snapshot` above. It's confusing to use the two terms interchangably when nowhere else in this document do we talk about `NodePool` at all.